src/main/java/formflow/library/ScreenController.java

```diff
@@ -196,8 +260,8 @@ ModelAndView postScreen(
     // Address validation
     handleAddressValidation(submission, formSubmission);
 
-    // handle submit actions, if requested
-    if (request.getRequestURI().toLowerCase().contains("submit")) {
+    // handle marking the record as submitted, if necessary
```

Comment:
Maybe do this as the last thing before right before save after the data merge from formData? Thoughts?

Reply:
I moved this down past the merge but before the action. I think it's still good to have this before the action.